### ISSUE TITLE
improve/reconcile bastion bootstrap method

### ISSUE THREAD

**Issue by wjordan** [aws-quickstart/quickstart-linux-bastion#1]
I came across Nicolas Malaval's [How to Record SSH Sessions Established Through a Bastion Host](https://aws.amazon.com/blogs/security/how-to-record-ssh-sessions-established-through-a-bastion-host/) 14 Jun 2016 post in the AWS Security Blog, which provides a [CloudFormation template](https://s3.amazonaws.com/awsiammedia/public/sample/RecordSSHSessionsBastionHost/template.json) for bootstrapping a bastion host. It seems that there's significant overlap between that blog post and this QuickStart solution, leading to some confusion as to which of the two solutions is the recommended approach, and what are the strengths/drawbacks or recommended use-cases for each.

Would it be possible to reconcile the two bastion-host bootstrap scripts, whether this involves deprecating one in favor of the other, or merging the features of both into a single unified and well-maintained solution?

Specifically, I'm interested in the specific set of commands contained within each approach's bootstrap shell script, which seem to be using completely different methods of locking down the bastion server. I'm wondering if a direct comparison can be made between them, and which approach is more secure, maintainable and/or battle-tested than the other.

**Comment by kamarja** [comment]
We have reviewed Nicolas' proposed bootstrap script and have decided to make a few modifications using his recommendations. For remote storage of the log files, we chose to use CloudWatch Logs instead of S3, since this is the kind of task for which it was designed. In the event of a network breach, the attacker will not have access rights to remove the CloudWatch logs. All SSH commands are logged and stored in the CloudWatch logs.  We have also added the option to disable/enable TCP and X11 Forwarding. They will both be disabled by default. Many thanks to Nicolas Malaval for his suggestions.

On the networking side, we have modified the architecture to use a configurable Auto-Scaling group (default is 1 and max instances is 4) which also uses Elastic IP addresses. This will help with maintaining high availability and consistency, in the case of failure of any of the Bastion Hosts.
